### PR TITLE
Code clean up for kubeadm

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Code clean up for kubeadm

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/apis/bootstraptoken/v1/utils.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ const (
 )
 
 // MarshalJSON implements the json.Marshaler interface.
-func (bts BootstrapTokenString) MarshalJSON() ([]byte, error) {
+func (bts *BootstrapTokenString) MarshalJSON() ([]byte, error) {
```
Comment (SataQiu) [inline cmd/kubeadm/app/apis/bootstraptoken/v1/utils.go:47]: Struct BootstrapTokenString has methods on both value and pointer receivers. Such usage is not recommended by the Go Documentation.

File: cmd/kubeadm/app/apis/bootstraptoken/v1/utils_test.go [hunk 2]
```diff
@@ -648,7 +648,7 @@ func TestBootstrapTokenFromSecret(t *testing.T) {
 					Name:      rt.name,
 					Namespace: "kube-system",
 				},
-				Type: v1.SecretType("bootstrap.kubernetes.io/token"),
+				Type: "bootstrap.kubernetes.io/token",
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/bootstraptoken/v1/utils_test.go:651]: this is also present on line 278, can we import 
`bootstrapapi "k8s.io/cluster-bootstrap/token/api"`

and use `bootstrapapi.SecretTypeBootstrapToken`?

File: cmd/kubeadm/app/cmd/upgrade/node.go [hunk 3]
```diff
@@ -206,7 +206,7 @@ func (d *nodeData) IgnorePreflightErrors() sets.Set[string] {
 	return d.ignorePreflightErrors
 }
 
-// KubeconfigPath returns the path to the user kubeconfig file.
+// KubeConfigPath returns the path to the user kubeconfig file.
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/upgrade/node.go:209]: strange, are we not running a linter on these packages - i.e. to be able to catch the godoc problems?

**Review by neolit123 — COMMENTED** [review]
LGTM, minus a couple of nits / questions.
thanks!

File: cmd/kubeadm/app/apis/bootstraptoken/v1/utils_test.go (hunk 2 above)
Comment (SataQiu) [reply cmd/kubeadm/app/apis/bootstraptoken/v1/utils_test.go]: Good idea! Fixed.

File: cmd/kubeadm/app/cmd/upgrade/node.go (hunk 3 above)
Comment (SataQiu) [reply cmd/kubeadm/app/cmd/upgrade/node.go:209]: It seems that the staticcheck's behavior is inconsistent with golint. 

https://github.com/golang/lint

Comment (SataQiu) [reply cmd/kubeadm/app/cmd/upgrade/node.go:209]:
```sh
$ cd cmd/kubeadm/app/constants
$ golint
constants.go:425:2: comment on exported const KubeControllerManagerUserName should be of the form "KubeControllerManagerUserName ..."
constants.go:427:2: comment on exported const KubeAPIServerUserName should be of the form "KubeAPIServerUserName ..."
constants.go:429:2: comment on exported const KubeSchedulerUserName should be of the form "KubeSchedulerUserName ..."
constants.go:431:2: comment on exported const EtcdUserName should be of the form "EtcdUserName ..."
constants.go:433:2: comment on exported const ServiceAccountKeyReadersGroupName should be of the form "ServiceAccountKeyReadersGroupName ..."

$ staticcheck
<no output>
```

Comment (neolit123) [reply cmd/kubeadm/app/cmd/upgrade/node.go:209]: hm, i was under the impression we are running a suite of linters on k/k, which covers the godoc exports:
https://github.com/kubernetes/kubernetes/blob/master/hack/verify-golangci-lint.sh

cc @pohly @dims

Comment (pohly) [reply cmd/kubeadm/app/cmd/upgrade/node.go:209]: We run golangci-lint with a pretty restricted set of enabled checks: https://github.com/kubernetes/kubernetes/blob/master/hack/golangci.yaml

See https://groups.google.com/a/kubernetes.io/g/dev/c/myGiml72IbM/m/aQTb6KToAwAJ for an effort to enable additional linters.

However, enforcing this particular godoc convention is not planned. It's too contentious and often not necessary.

Comment (neolit123) [reply cmd/kubeadm/app/cmd/upgrade/node.go:209]: thanks for the info, i guess we are going to have to run the godoc checks out of band / accept PRs that try to fix these.

Comment (pohly) [reply cmd/kubeadm/app/cmd/upgrade/node.go:209]: So you want "godocs" enabled for `cmd/kubeadm`?

Once https://github.com/golangci/golangci-lint/pull/3617 is merged, it should be possible to enable additional, stricter checks for specific directories that are not enabled elsewhere. Right now that is possible, but very awkward.

Comment (neolit123) [reply cmd/kubeadm/app/cmd/upgrade/node.go:209]: yes, we have to skip the kubeadm api packages, but it would be nice to enable it for the rest.

**Pull request merged**